src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -99,9 +100,27 @@ private void RequestRefresh(AuthenticationTicket ticket)
                 _refreshIssuedUtc = currentUtc;
                 var timeSpan = expiresUtc.Value.Subtract(issuedUtc.Value);
                 _refreshExpiresUtc = currentUtc.Add(timeSpan);
+                _refreshTicket = CloneTicket(ticket);
             }
         }
 
+        private AuthenticationTicket CloneTicket(AuthenticationTicket ticket)
+        {
```

Comment:
Does this introduce any sideeffects now that we are cloning the principal/identities?

Reply:
It's yet another failure point for people that want to use custom principal and identity types, but that's never worked anyways. Even custom identities would mostly work if they override Clone.